nodebuilder/header/header.go

```diff
@@ -29,22 +32,43 @@ func P2PExchange(cfg Config) func(params.Bootstrappers, host.Host) (header.Excha
 	}
 }
 
-// InitStore initializes the store.
-func InitStore(ctx context.Context, cfg Config, net params.Network, s header.Store, ex header.Exchange) error {
+// newSyncer constructs new Syncer for headers.
+func newSyncer(ex header.Exchange, store initStore, sub header.Subscriber, duration time.Duration) *sync.Syncer {
+	return sync.NewSyncer(ex, store, sub, duration)
+}
+
+// initStore is a type representing initialized header store.
+// NOTE: It is needed to ensure that Store is always initialized before Syncer is started.
```

Comment:
```suggestion
// initStore is a type representing initialized header store.
// NOTE:  It is needed to ensure a deterministic order for integration testing such that 
// Store is always initialized before the Syncer is started.
```

Reply:
This is not only for integration testing, btw. The ordering problem the initStore solves and the ordering problem of invokes are two different problems